### PR TITLE
Remove 'appBar' child component to fully support header styling

### PR DESCRIPTION
**Pull request by juhasuni**
This PR is related to #105 but instead of introducing two new props, we're adding full support for Header layout customisation using `style` prop.

Now, the header size can be also configured on per-screen basis as demonstrated here:

```
class FullScreen extends Component {
  static navigationOptions = {
    header: {
      style: {
        paddingTop: 0, // clears the default Header margin for the status bar
        height: 70 // sets new height for the Header
      }
    }
  }
}
```

### CODE REVIEW

File: src/views/Header.js [hunk 1]
```diff
@@ -279,26 +277,23 @@ class Header extends React.Component<void, HeaderProps, void> {
...
+        titleComponent,
+        leftComponent,
+        rightComponent
+      ];
     }
 
     return (
-      <Animated.View {...rest} style={[styles.container, style]}>
-        {appBar}
+      <Animated.View {...rest} style={[styles.container, {marginTop: STATUSBAR_HEIGHT, height: APPBAR_HEIGHT}, style]}>
```
Comment (satya164) [inline src/views/Header.js]: This should be padding, othewise the statusbar color won't match appbar

Comment (juhasuni) [inline src/views/Header.js]: Okay, I submitted a fix for this, which uses additional child component with `flex: 1`.

File: src/views/Header.js [hunk 2]
```diff
@@ -279,18 +277,16 @@ class Header extends React.Component<void, HeaderProps, void> {
...
+        titleComponent,
+        leftComponent,
+        rightComponent
+      ];
     }
 
     return (
       <Animated.View {...rest} style={[styles.container, style]}>
-        {appBar}
+        <View style={{flex: 1}}>{children}</View>
```
Comment (satya164) [inline src/views/Header.js]: Can you move `{ flex: 1 }` to `StyleSheet.create` below?

Comment (juhasuni) [inline src/views/Header.js]: Done

**Comment by grabbou** [comment]
I'd say we should wait for #257 to be merged, I've changed a lot in `render` method. Otherwise, I can rebase.

**Comment by satya164** [comment]
@grabbou since we're doing a release today, will be good to have this

**Comment by grabbou** [comment]
Cool, then just go ahead. I was busy with other OSS today, so haven't managed to finish on time. Let me know when the release gets cut and I'll follow up with a changelog.

**Comment by satya164** [comment]
Only if this CI things works. lol xD

**Comment by juhasuni** [comment]
Yup... it is nerve-wracking to work without CI and it is equally frustrating to try to keep them working.

**Pull request merged**